### DISCUSSION TITLE
Cmake can't find package

### DISCUSSION THREAD

**Discussion by HO-COOH** [alandefreitas/matplotplusplus#101] [Q&A] [answered]
**Bug category**

- [x] *bug - configure error* (I made it up)

**Describe the bug**
Cmake can't find the packages

**Steps to Reproduce**
1. Install the prebuilt binary (installer for 64bit windows), using the default install location ``C:\Program Files\matplotplusplus 1.0.1``
2. Use this cmake script
```cmake
cmake_minimum_required (VERSION 3.18)

project ("MatplotTest")

find_package(Matplot++ REQUIRED)

add_executable(Main MatplotTest.cpp)
target_link_libraries(Main PUBLIC Matplot++::matplot)
```

```bash
mkdir build
cd build
cmake .. 
```

**Output**

<details>
 
```console
C:\Users\Peter\source\repos\MatplotTest\build>cmake ..
-- The C compiler identification is MSVC 19.28.29515.1
-- The CXX compiler identification is MSVC 19.28.29515.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Preview/VC/Tools/MSVC/14.28.29515/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Preview/VC/Tools/MSVC/14.28.29515/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:8 (find_package):
  By not providing "FindMatplot++.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Matplot++", but CMake did not find one.

  Could not find a package configuration file provided by "Matplot++" with
  any of the following names:

    Matplot++Config.cmake
    matplot++-config.cmake

  Add the installation prefix of "Matplot++" to CMAKE_PREFIX_PATH or set
  "Matplot++_DIR" to a directory containing one of the above files.  If
  "Matplot++" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "C:/Users/Peter/source/repos/MatplotTest/build/CMakeFiles/CMakeOutput.log".
```
</details>

**Platform**

- [ ] *cross-platform issue - linux*
- [x] *cross-platform issue - windows*
- [ ] *cross-platform issue - macos*

**Environment Details:**
 - OS: Windows 10
 - OS Version: 2004
 - Compiler: Visual Studio
 - Compiler version: v18.1

**Comment by alandefreitas** [comment, marked as answer]
https://stackoverflow.com/questions/21314893/what-is-the-default-search-path-for-find-package-in-windows-using-cmake

https://alandefreitas.github.io/matplotplusplus/integration/binary-packages.html

```cmake
list(APPEND CMAKE_MODULE_PATH put/your/installation/directory/here)
find_package(Matplot++ REQUIRED)
# ...
target_link_libraries(my_target PUBLIC Matplot++::matplot)
```

**Reply by HO-COOH** [reply]
Yes, so according to the documentation:
>CMake should be able to locate the matplot++-config.cmake script automatically if you installed the library under /usr/local/ (Linux / Mac OS) or C:/Program Files/ (Windows). 

But it does not.
Also these doesn't seem to work as well.
```cmake
list(APPEND CMAKE_MODULE_PATH "C:\\Program Files\\matplotplusplus 1.0.1")
list(APPEND CMAKE_MODULE_PATH "C:\\Program Files\\matplotplusplus 1.0.1\\lib")
list(APPEND CMAKE_MODULE_PATH "C:\\Program Files\\matplotplusplus 1.0.1\\lib\\cmake\\Matplot++")
```
But this work
```cmake
set(Matplot++_DIR "C:\\Program Files\\matplotplusplus 1.0.1\\lib\\cmake\\Matplot++")
```

**Reply by HO-COOH** [reply]
@alandefreitas Maybe using ``set`` is not the intended usage. And if you can reproduce this, I suggest you move this back to issue, because it is indeed a documentation issue :)

**Reply by alandefreitas** [reply]
Yes. This is a issue:

> Yes, so according to the documentation:
> 
> > CMake should be able to locate the matplot++-config.cmake

The initial issue was more like a question because it still wasn't considering all aspects of the documentation and wasn't related to the code. So I moved it here so that more people could help.

Unfortunately, discussions cannot become issues again, but I've already adjusted the documentation with your recommendations. Feel free to open another more specific documentation issue or a PR changing the docs.
